extensions/common/context.cc

```diff
@@ -295,13 +269,69 @@ flatbuffers::DetachedBuffer extractLocalNodeFlatBuffer() {
   return fbb.Release();
 }
 
+PeerNodeInfo::PeerNodeInfo(const std::string_view peer_metadata_id_key,
+                           const std::string_view peer_metadata_key) {
+  found_ = getValue({peer_metadata_id_key}, &peer_id_);
+  if (found_) {
+    getValue({peer_metadata_key}, &peer_node_);
+    return;
+  }
+
```

Comment:
If downstreamPeer { return }

No need to check upstream host metadata here.

Reply:
Done.